app/helpers/lookup_keys_helper.rb

```diff
@@ -68,53 +68,123 @@ def validator_type_selector(f)
-    if no_value && obj.class.model_name == "Host"
-      if key.required
-        explanation.prepend(_("Required parameter without value.<br/><b>Please override!</b><br/>"))
-        icon = "warning-sign"
-      else
-        explanation.prepend(_("Optional parameter without value.<br/><i>Won't be given to Puppet.</i><br/>"))
-        icon = "exclamation-sign"
-      end
+  def override_toggle(overridden)
+    link_to_function(icon_text('edit'), "override_class_param(this)",
+                     :title => _("Override this value"),
+                     :'data-tag' => 'override',
+                     :class =>"btn btn-default btn-md #{'hide' if overridden}") +
+      link_to_function(icon_text('remove'), "override_class_param(this)",
+                       :title => _("Remove this override"),
+                      :'data-tag' => 'remove',
+                      :class =>"btn btn-default btn-md #{'hide' unless overridden}")
+  end
+
+  def lookup_value(host_or_hostgroup, lookup_key)
+    lookup_key.overridden_value(host_or_hostgroup) || LookupValue.new
+  end
+
+  def class_parameters_table_title(puppetclass_name, rowspan)
+    content_tag(:td, puppetclass_name, :rowspan => rowspan, :class => "ellipsis")
+  end
+
+  def use_puppet_default_check_box(lookup_key, lookup_value, disabled)
+    return unless lookup_key.type == "PuppetclassLookupKey"
+    check_box(lookup_value_name_prefix(lookup_key.id), :use_puppet_default,
+              :value    => lookup_value.id,
+              :disabled => disabled,
+              :onchange => "toggleUsePuppetDefaultValue(this, 'value')",
+              :hidden   => disabled,
+              :title    => _('Use Puppet default'),
+              :checked  => lookup_value.use_puppet_default)
+  end
+
+  def remove_lookup_value_fields(lookup_key, lookup_value, disabled)
+    return unless can_edit_params?
+    value_prefix = lookup_value_name_prefix(lookup_key.id)
+    hidden_field(value_prefix, :lookup_key_id, :value => lookup_key.id,
+                 :disabled => disabled, :class => 'send_to_remove') +
+      hidden_field(value_prefix, :id, :value => lookup_value.id,
+                   :disabled => disabled, :class => 'send_to_remove') +
+        hidden_field(value_prefix, :_destroy, :value => false,
+                     :disabled => disabled, :class => 'send_to_remove destroy')
+  end
+
+  # Input tags used to override lookup keys need a 'name' HTML attribute to
+  # tell Rails which lookup_value they belong to.
+  # This method returns the name attribute for any combination of lookup_key
+  # and host/hostgroup. Other objects that may receive parameters too will need
+  # to override this method in their respective helpers.
+  def lookup_value_name_prefix(lookup_key_id)
+    "#{parameters_receiver}[lookup_values_attributes][#{lookup_key_id}]"
+  end
+
+  def parameters_receiver
+    return 'host' if params[:host].present? || params[:controller] == 'hosts'
```

Comment:
Kind of limited in case a new model in a plugin starts using parameters too, that's why I said a Parameters aspect (https://github.com/theforeman/foreman/pull/2665) would be helpful.